### DISCUSSION TITLE
convert a:png is giving error. why?

### DISCUSSION THREAD

**Discussion by Sadiq3210** [ImageMagick/ImageMagick#1905] [General]
`convert a.png -crop 40x30+10+10 a40-3-0-10-10.png`
is working fine. but

`convert a: -crop 40x30+10+10 a40-3-0-10-10.png` results in error
```
convert: must specify image size `' @ error/raw.c/ReadRAWImage/140.
convert: no images defined `a40-3-0-10-10.png' @ error/convert.c/ConvertImageCommand/3282.
```

**Comment by fmw42** [comment]
a: is not an image

**Reply by Sadiq3210** [reply]
I am trying to generate examples from this website [https://www.imagemagick.org/Usage/crop/](https://www.imagemagick.org/Usage/crop/)

```
convert rose:                    rose.gif   
convert rose: -crop 40x30+10+10  crop.gif   
convert rose: -crop 40x30+40+30 crop_br.gif   
convert rose: -crop 40x30-10-10  crop_tl.gif   
convert rose: -crop 90x60-10-10  crop_all.gif   
convert rose: -crop 40x30+90+60  crop_miss.gif
```
What does rose: implies here?

**Comment by fmw42** [comment]
rose: is an internal Imagemagick image. It includes the : and has no suffix. See https://imagemagick.org/script/formats.php#builtin-images  All built-in images end in : No other image use the colon. They must have proper suffix.
